src/limit_resource.rs

```diff
@@ -165,6 +170,17 @@ pub fn limit_resource(m: &mut Module, config: &Config) {
     });
 }
 
+fn limit_heap_memory(m: &mut Module, limit: u32) {
+    // mqulti-memory is disabled at compilation
+    if let Some(memory) = m.memories.iter_mut().next() {
```

Comment:
```suggestion
    assert_eq!(m.memories.len(), 1);
    let memory = m.memories[0];
```

Reply:
Switched to [`get_memory_id`](https://docs.rs/walrus/latest/walrus/struct.Module.html#method.get_memory_id) which provides the same guarantee as the assert.